prymer/primer3/primer3_parameters.py

```diff
@@ -103,10 +154,15 @@ def __post_init__(self) -> None:
         if self.gc_clamp[0] > self.gc_clamp[1]:
             raise ValueError("Min primer GC-clamp must be <= max primer GC-clamp")
 
+        # Set melting temperature thresholds to be 10 degrees less than the minimum primer tm
+        default_thermo_tm: float = self.primer_tms.min - 10.0
+        object.__setattr__(self, "primer_max_self_any_thermo", default_thermo_tm)
+        object.__setattr__(self, "primer_max_self_end_thermo", default_thermo_tm)
+        object.__setattr__(self, "primer_max_hairpin_thermo", default_thermo_tm)
```

Comment:
Why not just do this above?

Reply:
At this point in the code, you have validated other user inputs and you know that you have valid `PrimerAndAmpliconParameters`. It made more sense to me to set these attributes after you know that you have valid input and can continue.